File: coding/18th September 2025/Count Negative Numbers in a Sorted Matrix.java
//Count Negative Numbers in a Sorted Matrix
class Solution 
{
    public int countNegatives(int[][] grid) 
    {
        int result = 0;
        for(int i = 0; i < grid.length; i++)
        {
            int j = grid[0].length - 1;
            if(grid[i][j] < 0)
            {
                while(j >= 0 && grid[i][j] < 0)
                {
                    j--;
                    result++;
                }
            }
            else
                continue;
        }
        return result;
    }
}

File: Coding in Python/28th December 2025/Count Negative Numbers in a Sorted Matrix.py
#Count Negative Numbers in a Sorted Matrix
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        result = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if(grid[i][j] < 0):
                    result += len(grid[i]) - j
                    break
        return result
